### PR TITLE
[PM-4767] Update Grant_Save procedure

### PR DESCRIPTION
**Pull request by rkac-bw**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Improve concurrency with `Grant_Save` stored procedure. Updates the stored procedure to use update insert instead of merge to avoid locking issues with merge.

## Code changes

* **util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql / src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql:** "update, insert" instead of "merge".

### CODE REVIEW

File: src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ﻿CREATE PROCEDURE [dbo].[Grant_Save]
-    @Key  NVARCHAR(200),
+  @Key  NVARCHAR(200),
```
Comment (withinfocus) [inline src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql:2]: ⛏️ Go ahead and open this file in an editor with Prettier support. I prefer VS Code. It'll keep this straight.

Separately, I feel our SQL formatting isn't really defined and we could work on that e.g. why are two spaces after some of these parameter names allowed?

File: src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql [hunk 2]
```diff
@@ -12,53 +12,26 @@
 AS
 BEGIN
     SET NOCOUNT ON
+	--select 1
```
Comment (withinfocus) [inline src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql:15]: ⛏️ Clean up.

File: util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @ClientId NVARCHAR(200),
+    @Description NVARCHAR(200),
+    @CreationDate DATETIME2,
+    @ExpirationDate DATETIME2,
+    @ConsumedDate DATETIME2,
+    @Data NVARCHAR(MAX)
+AS
+BEGIN
+    SET NOCOUNT ON
+	--select 1
```
Comment (withinfocus) [inline util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql:15]: ⛏️ Be sure to copy your changes above here.

**Review by withinfocus — CHANGES_REQUESTED** [review]
Couple minor things.

File: src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql [hunk 4]
```diff
@@ -1,61 +1,57 @@
...
+	UPDATE [dbo].[Grant]
+	SET [Type] = @Type,
+		[SubjectId] = @SubjectId,
+		[SessionId] = @SessionId,
+		[ClientId] = @ClientId,
+		[Description] = @Description,
+		[CreationDate] = @CreationDate,
+		[ExpirationDate] = @ExpirationDate,
+		[ConsumedDate] = @ConsumedDate,
+		[Data] = @Data
```
Comment (withinfocus) [inline src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql:26]: ⛏️ Spacing is weird to me here.

File: util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql [hunk 5]
```diff
@@ -1,61 +1,54 @@
...
-    @Type  NVARCHAR(50),
-    @SubjectId NVARCHAR(200),
-    @SessionId NVARCHAR(100),
-    @ClientId NVARCHAR(200),
-    @Description NVARCHAR(200),
-    @CreationDate DATETIME2,
-    @ExpirationDate DATETIME2,
-    @ConsumedDate DATETIME2,
-    @Data NVARCHAR(MAX)
+CREATE OR ALTER PROCEDURE [dbo].[Grant_Save] @Key NVARCHAR(200),
```
Comment (withinfocus) [inline util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql:1]: ⛏️ Newline for the first parameter -- agree?

File: src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql (hunk 2 above)
Comment (rkac-bw) [reply src/Sql/Auth/dbo/Stored Procedures/Grant_Save.sql]: Resubmitted

File: util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql (hunk 5 above)
Comment (rkac-bw) [reply util/Migrator/DbScripts/2024-01-03_00_FixGrantSave.sql]: Yes, resubmitted

**Review by jlf0dev — APPROVED** [review]

**Review by withinfocus — APPROVED** [review]
I felt like I was being annoying and unspecific with the formatting, so I opened this up and ran Prettier on it so you can see what I mean.

**Review by jlf0dev — APPROVED** [review]

**Review by withinfocus — APPROVED** [review]

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**